monitor/src/main/java/edu/asu/diging/monitor/core/service/impl/AppHelper.java

```diff
@@ -32,6 +37,7 @@ public IApp copyAppInfo(IApp app, AppForm appForm) {
         if (appForm.getRecipientIds() != null) {
             app.setRecipients(getRecipientsById(appForm.getRecipientIds()));
         }
+        appManager.encryptPassword(appForm, app);
```

Comment:
this should be done in teh addapp method of the manager. think about it this way: this always needs to happen when an app is added or modified. if you create the app in a different way than through the copyAppInfo method, you always have to remember to encode the password in the same way. It's likely to forget this. if the appManager takes care of it, you (or whoever develops monitor) don't have to worry about this at all.